control-plane/csi-driver/src/context.rs

```diff
@@ -410,6 +420,25 @@ impl TryFrom<&HashMap<String, String>> for PublishParams {
     }
 }
 
+/// Validate the topology parameters.
+/// Errors out of nodeSpreadTopologyKey has same values as of
+/// node_has_topology_key/nodeAffinityTopologyLabel
+pub(crate) fn validate_topology_params(
+    map1: Option<HashMap<String, String>>,
+    map2: Option<HashMap<String, String>>,
+) -> Result<(), tonic::Status> {
+    if let (Some(map1), Some(map2)) = (map1, map2) {
+        let has_common_keys = map1.len() == map2.len() && map1.keys().all(|k| map2.contains_key(k));
```

Comment:
Shouldn't it be any?

```suggestion
        let has_common_keys = map1.len() == map2.len() && map1.keys().any(|k| map2.contains_key(k));
```